torchmultimodal/modules/encoders/swin_transformer_3d_encoder.py

```diff
@@ -0,0 +1,567 @@
+# Modified from 2d Swin Transformers in torchvision:
+# https://github.com/pytorch/vision/blob/main/torchvision/models/swin_transformer.py
+
+from functools import partial, lru_cache
+from typing import Optional, Callable, List, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+from torchvision.models.vision_transformer import MLPBlock
+from torchvision.ops.stochastic_depth import StochasticDepth
+
+
+def _compute_pad_size_3d(
```

Comment:
why do u need this and next function to be global?

Reply:
For `_compute_pad_size_3d` it is needed on 2 different classes so we put it on global but private.
For `_compute_attention_mask_3d` we put this as function so we can cache it. We don't put it inside the function `shifted_window_attention_3d` because we avoid nested function as it is not supported by jit script.